src/bun.js/api/server.zig

```diff
@@ -5297,6 +5304,44 @@ pub fn NewServer(comptime NamespaceType: type, comptime ssl_enabled_: bool, comp
             return JSC.JSValue.jsNumber(@as(i32, @intCast(@as(u31, @truncate(this.activeSocketsCount())))));
         }
 
+        pub fn getAddress(this: *ThisServer, globalThis: *JSGlobalObject, _: *JSC.CallFrame) callconv(.C) JSC.JSValue {
+            switch (this.config.address) {
+                .unix => |unix| {
+                    var value = bun.String.create(bun.sliceTo(@constCast(unix), 0));
+                    defer value.deref();
+                    return value.toJS(globalThis);
+                },
+                .tcp => {
+                    if (this.cached_localAddr.address.isEmpty()) {
+                        if (this.listener) |listener| {
+                            var buf: [64]u8 = [_]u8{0} ** 64;
+                            var len: i32 = 64;
+                            var is_v4: bool = false;
+                            listener.socket().localAddressText(&buf, &len, &is_v4);
+                            if (len > 0) {
+                                this.cached_localAddr = .{
+                                    .address = bun.String.create(buf[0..@as(usize, @intCast(len))]),
+                                    .is_v4 = is_v4,
+                                };
+                            }
+                        }
+                    }
+                    if (this.cached_localAddr.address.isEmpty()) {
+                        return JSValue.jsNull();
+                    }
+                    var result = JSC.JSValue.createEmptyObject(globalThis, 3);
```

Comment:
if you look at the requestIP, specficically `JSSocketAddress.cpp` code there is already a binding and structure to create this object, but as jarred said the performance of this isn't that much of a concern

Reply:
Ok, I reuse the `JSSocketAddress` class.